### PR TITLE
Shanoir issue#2097 uploader from folder

### PR DESCRIPTION
**Pull request by jcomedouteau**
Not ready yet.
The idea of this feature is to allow the uploading of dicom data from a folder into shanoir using shanoir-uploader.

The folder structure should be the following:

Parent folder
 - subject_name_1
    - exam_comment_1
      - [dcm files with potential subfolders]
    - examn_comment_2
      - [dcm files with potential subfolders]
 - subect_name_2
    - exam_comment_1
      - [dcm files with potential subfolders]

### CODE REVIEW

**Comment by michaelkain** [comment]
Dear @jcomedouteau, I hope you are doing fine. I am looking forward to review this PR.
We would like to experiment CSV import from pacs with OFSEP in Rennes, to mass import from the pacs.
Do you think you could test this feature on this new PR? as maybe similar codes used? I think we should
finish this work + my PR with my latest changes (OSIV false) to have a new code base before starting the tests with OFSEP in Rennes. What do you think? With kind regards, Michael

**Comment by michaelkain** [comment]
Hi @jcomedouteau, I hope you are doing fine. If possible, it might be great to handle it to me before your holidays, as I am contacted now every week to set up mass imports with sh-up (csv-pacs) and I would like to have one code base, see my PR OSIV (latest devs on ShUp). With kind regards, Michael

**Comment by jcomedouteau** [comment]
Hi @michaelkain, I think we are ready for now.

File: shanoir-uploader/src/main/java/org/shanoir/uploader/action/UploadFromFolderActionListener.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+    JFileChooser fileChooser;
+    ImportFromFolderWindow importFromFolderWindow;
+    private ResourceBundle resourceBundle;
+
+    private static Logger logger = Logger.getLogger(UploadFromCsvActionListener.class);
+
+    public UploadFromFolderActionListener(ImportFromFolderWindow importFromFolderWindow, ResourceBundle resourceBundle) {
+        this.importFromFolderWindow = importFromFolderWindow;
+        this.fileChooser = new JFileChooser("/home/jcome/Documents/Shanoir/Data");
```
Comment (michaelkain) [inline shanoir-uploader/src/main/java/org/shanoir/uploader/action/UploadFromFolderActionListener.java:35]: Hi @jcomedouteau, thank you for this great PR! Absolute path to your folder, I will change this on develop.

File: shanoir-uploader/src/main/java/org/shanoir/uploader/action/UploadFromFolderActionListener.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        this.importFromFolderWindow.studyCardCB.setEnabled(false);
+
+        FolderImport folderImport = new FolderImport();
+        folderImport.setStudy(((Study)importFromFolderWindow.studyCB.getSelectedItem()));
+        folderImport.setStudyCard(((StudyCard)importFromFolderWindow.studyCardCB.getSelectedItem()));
+
+        folderImport.setExaminationImports(new ArrayList<>());
+        // Check if it is a folder
+        if (!selectedFile.isDirectory()) {
+            this.importFromFolderWindow.displayError("ARG, file is not a folder");
```
Comment (michaelkain) [inline shanoir-uploader/src/main/java/org/shanoir/uploader/action/UploadFromFolderActionListener.java:67]: I will change the ARGs

File: shanoir-uploader/src/main/java/org/shanoir/uploader/action/ImportFromFolderRunner.java [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+            this.importFromFolderWindow.frame.dispose();
+        } else {
+            this.importFromFolderWindow.displayImports(folderimport);
+            importFromFolderWindow.openButton.setEnabled(true);
+            importFromFolderWindow.uploadButton.setEnabled(false);
+        }
+        return null;
+    }
+
+    private boolean importData(ExaminationImport importTodo) {
```
Comment (michaelkain) [inline shanoir-uploader/src/main/java/org/shanoir/uploader/action/ImportFromFolderRunner.java:108]: Hi @jcomedouteau, I will split your method importData, which is very long.

**Review by michaelkain — APPROVED** [review]
Hi @jcomedouteau, thank you for this great PR again!
I approve it, as it is a big extension on the road to mass
imports. I will align some pieces with CSV and standard
imports, e.g. the profile is not respected today for pseudo
creation, so your code does not yet work for OFSEP.
The importData method should be factored and maybe
we can avoid to manipulate external GUI components in
this method as well, to see. Thank you again and with kind
regards, Michael

**Pull request merged**